### PR TITLE
Zip as possible Zarr extenstion

### PR DESCRIPTION
**Pull request by aulemahal**
### Pull Request Checklist:
- [ ] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #xyz
- [x] (If applicable) Documentation has been added / updated (for bug fixes / features).
- [ ] (If applicable) Tests have been added.
- [x] This PR does not seem to break the templates.
- [x] CHANGELOG.rst has been updated (with summary of main changes).
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added.

### What kind of change does this PR introduce?

* The directory parser will now assign a "zarr" format to files with .zarr.zip or .zip extensions. The extension still needs to be included in the pattern.
* `get_engine` returns `zarr` for the same two file extensions.

### Does this PR introduce a breaking change?
No.

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -16,6 +16,7 @@ Internal changes
 ^^^^^^^^^^^^^^^^
 * Include domain in `weight_location` in ``regrid_dataset``. (:pull:`414`).
 * Added pins to `xarray`, `xclim`, `h5py`, and `netcdf4`. (:pull:`414`).
+* Add ``.zip`` and ``.zarr.zip`` as possible file extensions for Zarr datasets. (:pull: `426`).
```
Comment (Zeitsperre) [inline CHANGELOG.rst:19]:
```suggestion
* Add ``.zip`` and ``.zarr.zip`` as possible file extensions for Zarr datasets. (:pull:`426`).
```

File: src/xscen/catutils.py [hunk 2]
```diff
@@ -39,6 +39,12 @@
 # ## File finding and path parsing ## #
 
 
+SUFFIX_TO_FORMAT = {".nc": "nc", ".zip": "zarr", ".zarr.zip": "zarr", ".zarr": "zarr"}
```
Comment (Zeitsperre) [inline src/xscen/catutils.py:42]: I know it's rarer these days, but I'd also add `.nc4` to the mix.

File: src/xscen/catutils.py [hunk 3]
```diff
@@ -223,7 +229,7 @@ def _name_parser(
         return None
 
     d["path"] = abs_path
-    d["format"] = path.suffix[1:]
+    d["format"] = SUFFIX_TO_FORMAT.get(path.suffix)
```
Comment (Zeitsperre) [inline src/xscen/catutils.py:232]: Is `d["format"] = None` possible/covered?

**Review by Zeitsperre — APPROVED** [review]

File: src/xscen/catutils.py (hunk 3 above)
Comment (aulemahal) [reply src/xscen/catutils.py:232]: hmm I guess not, I'll put the previous value as default.

**Pull request merged**